--- src/Makefile	2016-11-30 09:59:12.889526296 -0800
+++ src/Makefile	2016-11-30 10:00:46.812341855 -0800
@@ -2,12 +2,12 @@
 includedir=$(prefix)/include
 libdir=$(prefix)/lib

-CFLAGS ?= -g -fomit-frame-pointer -O2
-CFLAGS += -nostdlib -nostartfiles -Wall -I. -fPIC
-SO_CFLAGS=-shared $(CFLAGS)
-L_CFLAGS=$(CFLAGS)
-LINK_FLAGS=
-LINK_FLAGS+=$(LDFLAGS)
+MK_CPPFLAGS=-I.
+MK_CFLAGS=-nostdlib -nostartfiles -fomit-frame-pointer -fPIC $(CFLAGS)
+CFLAGS=-Wall -g -O2
+SO_CFLAGS=-shared $(MK_CFLAGS)
+L_CFLAGS=$(MK_CFLAGS)
+MK_LDFLAGS=-Wl,--as-needed -lgcc -lc $(LDFLAGS)

 soname=libaio.so.1
 minor=0
@@ -37,10 +37,10 @@
 $(libaio_objs) $(libaio_sobjs): libaio.h vsys_def.h

 %.os: %.c
-	$(CC) $(SO_CFLAGS) -c -o $@ $<
+	$(CC) $(MK_CPPFLAGS) $(SO_CFLAGS) -c -o $@ $<

 %.ol: %.c
-	$(CC) $(L_CFLAGS) -c -o $@ $<
+	$(CC) $(MK_CPPFLAGS) $(L_CFLAGS) -c -o $@ $<

 AR ?= ar
 RANLIB ?= ranlib
@@ -50,7 +50,7 @@
 	$(RANLIB) libaio.a

 $(libname): $(libaio_sobjs) libaio.map
-	$(CC) $(SO_CFLAGS) -Wl,--version-script=libaio.map -Wl,-soname=$(soname) -o $@ $(libaio_sobjs) $(LINK_FLAGS)
+	$(CC) $(SO_CFLAGS) -Wl,--version-script=libaio.map -Wl,-soname=$(soname) -o $@ $(libaio_sobjs) $(MK_CPPFLAGS)

 install: $(all_targets)
 	install -D -m 644 libaio.h $(includedir)/libaio.h
